### ISSUE TITLE
Got it working, any suggestions?

### ISSUE THREAD

**Issue by akarkou1** [Nem4/viewerbot#77]
I got everything updated and working, I fixed the multi threading issue in which the max threads was being completely ignored, I also added an argument that wasn't included although it was mentioned (to limit threads). Any other suggestions on what I should get done with this project before moving on (releasing it publicly)?

**Comment by Anonymous2** [comment]
check issues #74 and #76

**Comment by akarkou1** [comment]
> check issues #74 and #76

Issues from 74: as far as tor proxies, I can almost guarantee that there would be issues connecting to twitch with them,
issues from #76 as far as checking proxies goes it would be very easy to just check proxies prior or during the process however the real problem would be with proxies dropping during the botting. for this we can either completely eliminate the proxy from the proxies file or end the thread for that specific proxy. both should be easy, I can also add a pre-run proxy check however no promises on that.

**Comment by Nem4** [comment]
Hello there, @akarkou1! Thanks for your interest in this project. I keep getting emails from GitHub and just can't ignore them anymore.

Can you answer a few questions for me?

- Does it really work? 
- How stable is it? 
- Is it possible to precisely control (+-5%) the number of viewers?
- Are public proxies work or just private ones?

Thanks again for your interest.

I stopped working on the project because I lost the need for it. 
I would really appreciate If you can make it work, because I keep getting messages from people asking me to resurrect the project.

**Comment by akarkou1** [comment]
> 

stability seems to be an issue, however I haven't debugged it further than making sure streamlink views still counted towards views, as far as precise control, yes, it does seem like with a good proxy list you're able to control the amount of views fairly well (could be improved by refreshing every once in a while). A majority of my testing was done on private/shared proxies, however it seems like the bandwidth usage is extremely minimal, meaning anyone should be able to shell out the 2-10 bucks per gb and have hours worth of botting time. I'm going to keep working on it further and updating as I go.
edit 1: This method certainly seems a lot more viable than running 100x selenium instances. I managed to get my viewer count to 100 with no problem using minimal resources.

**Comment by Nem4** [comment]
@akarkou1 can you make a pull request? I'm curious what you did there

**Comment by 9ofe** [comment]
@akarkou1 , When do you intend to release it?

**Comment by akarkou1** [comment]
> @akarkou1 can you make a pull request? I'm curious what you did there



> @akarkou1 , When do you intend to release it?

I will when it's all completely done, right now I'm testing with VLC and it seems to be working pretty damn well, the problem with releasing that to the public right now is that there would be a installation process to everything for it to work. however I'm sure I can figure something out.

**Comment by msdart** [comment]
> > @akarkou1 can you make a pull request? I'm curious what you did there
> 
> > @akarkou1 , When do you intend to release it?
> 
> I will when it's all completely done, right now I'm testing with VLC and it seems to be working pretty damn well, the problem with releasing that to the public right now is that there would be a installation process to everything for it to work. however I'm sure I can figure something out.

You can just release what you have now. I think the community will help with everything else.

**Comment by Chel-Jessen** [comment]
Yeah I'm really looking forward to work on this project but to avoid duplicate work I'll wait until @akarkou1 releases his version

**Comment by akarkou1** [comment]
Here is a quick update: I haven't really been working on it as school just started, however I might have found a method to literally start this project from the start. I'm going to do some testing later this week but as far as I know it looks like the only way most commercial bots are running. will keep y'all updated later this weekend.

**Comment by akarkou1** [comment]
by the looks of things, it's not gonna be a very easy thing to do alone. I've tried launching different vlc instances to see if the views would still be counted however that doesn't seem to be the case. I'm going to try a few things later this week before completely giving up.

**Comment by Chel-Jessen** [comment]
can u do a fork or repo with your current progress?

**Comment by SRZIsHere** [comment]
Hello @akarkou1 : Is that possible to have your Discord or anything to communicate with you ?
